packages/astro/test/fixtures/lit-element/src/components/my-element.js

```diff
@@ -3,9 +3,26 @@ import { LitElement, html } from 'lit';
 export const tagName = 'my-element';
 
 export class MyElement extends LitElement {
+  static properties = {
+    bool: {type: Boolean},
+    str: {type: String},
+    obj: {type: Object},
```

Comment:
The type parameters only matter when serializing and deserializing from an attribute. If might be good to double-check both that and that reflecting properties work by adding a `reflect: true` property. Maybe on a Boolean property.

Reply:
`reflect: true` will not reflect back at server time unless we call `connectedCallback` on the server. The problems are:

1. I don't know if Astro calls this function only when it's ready to mount or every time @matthewp
2. if it's a boolean, lit-ssr will serialize the attribute to `reflectedBool='false'` which runs us into the exact same issue as before. Is this a lit-ssr bug?

Reply:
`connectedCallback` is not called by this renderer. It defers everything to Lit's SSR rendering capabilities which I believe only calls the constructor.

Reply:
Ah, so in this case we should be calling connectedCallback on the server, because if it's going through that generator it's connected to the page, right?

Reply:
Added connectedCallback call, but did not add test for reflected boolean until this issue is resolved:

lit/lit#2305

Reply:
update, this is actually a non-issue and already solved by my PR. pretty sure I wrote this before I made the patch to call `instance.connectedCallback()`. Working on a PR to add tests for this